### DISCUSSION TITLE
Runtime error 401 image.

### DISCUSSION THREAD

**Discussion by arumenoy** [twinbasic/twinbasic#1658] [General]
Hello, I have an error with the software "vbmodal". The software screen appears. But I get a runtime error when I "show" or "vbmodal" to show other windows, or for some other reason.Are non-English character codes supported?
Is this a feature that has not been implemented yet? I am looking forward to the completion of TwinBasic.
![error1](https://github.com/twinbasic/twinbasic/assets/135419894/7d874fc6-b41d-4637-a93c-9a66f9c60b66)
![error2](https://github.com/twinbasic/twinbasic/assets/135419894/dc122c35-3026-4407-bdde-e6b10d90e391)

**Comment by sokinkeso** [comment]
If you try to open a non-modal Form from a modal Form, then this runtime error is normal.
Same behavior have vb6.
See the following example:
form1 : non-modal (start form)
form2: modal
form3: non-modal

If you open a modal Form, then all other Forms that will open from this Form, have to be modal too.
The same error will occur if you open form2 as modal (from Form1), and have a timer event firing in form1, that will try to show form3 as non-modal.

![image](https://github.com/twinbasic/twinbasic/assets/113173954/226ba05b-e6e5-43bf-9b97-0bcfd311864c)

**Reply by arumenoy** [reply]
Hmmm...I am testing with software that works without errors in VB6.0 (SP6/Windows XP).The binary (exe) file is also working on Windows 11. I am sorry, but ......TwinBasic is not fully compatible with VB6.0.

I have TwinBasic loaded and running a VB6.0 project file.When I try to display a modal form from a modal form, I get an error like the one in the image.Non-modal is a software that I almost never use, so maybe there is a different factor.

**Reply by sokinkeso** [reply]
https://github.com/twinbasic/twinbasic/assets/113173954/452deb14-bee1-4de7-9087-c4dd63861d89

Creating a new project, modal forms seems to work ok. I don't know if something else of your code is causing the error.
As of the compatibility with vb6, don't forget that TwinBasic is still in BETA state..
I believe Wayne will have an answer about this.

**Reply by arumenoy** [reply]
I am an engineer, so I understand that this is a beta version. If using an existing project file causes an error, then it is still difficult for it to become a practical development tool. Thanks.
